src/app/forms/validators/custom-async-validators.ts

```diff
@@ -114,7 +117,62 @@ export class CustomAsyncValidators {
 
           return null;
         })
+   
+        );
+  }
+
+  static hideIfEqualsWithCondition(store: Store<State>, sibling: string, value: string, conditions: Condition | Condition[]): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> =>
+      store.pipe(
+        take(1),
+        select(testResultInEdit),
+        map(testResult => {
+
+          if(testResult){
+            const conditionsPassed = CustomAsyncValidators.checkConditions(testResult, conditions);
+
+            if (control?.parent) {
+              const siblingControl = control.parent.get(sibling) as CustomFormControl;
+              siblingControl.meta.hide = Array.isArray(value) ? (conditionsPassed && value.includes(control.value)) : (conditionsPassed && control.value === value);
+            }
+          }
+
+          return null;
```

Comment:
This can be clarified.
```suggestion
          if (!testResult || !control?.parent) { return null; }

          const siblingControl = control.parent.get(sibling) as CustomFormControl;

          const conditionsPassed = CustomAsyncValidators.checkConditions(testResult, conditions);

          siblingControl.meta.hide = conditionsPassed && (Array.isArray(value) ? value.includes(control.value) : control.value === value);
```

Reply:
Thanks Taha that looks cleaner I will commit that.